### ISSUE TITLE
oemof does not work with couenne and bonmin as solvers

### ISSUE THREAD

**Issue by longlevan** [oemof/oemof-solph#774]
Hello, 
I am a newbie for oemof-solph. I want to test MINLP solvers like bonmin, couenne with oemof-solph. However,  I got the following error:

```
RuntimeError                              Traceback (most recent call last)
<ipython-input-38-f9ef7be408b1> in <module>
     10 # if tee_switch is true solver messages will be displayed
     11 logging.info("Solve the optimization problem")
---> 12 om.solve(solver="bonmin", solve_kwargs={"tee": True})

g:\Workspace\Cenaero\Git\h2cs-design\env\lib\site-packages\oemof\solph\models.py in solve(self, solver, solver_io, executable, **kwargs)
    215             options[k] = solver_cmdline_options[k]
    216 
--> 217         solver_results = opt.solve(self, **solve_kwargs)
    218 
    219         status = solver_results["Solver"][0]["Status"]

g:\Workspace\Cenaero\Git\h2cs-design\env\lib\site-packages\pyomo\opt\base\solvers.py in solve(self, *args, **kwds)
    104     def solve(self, *args, **kwds):
    105         """Perform optimization and return an SolverResults object."""
--> 106         self._solver_error('solve')
    107 
    108     def reset(self):

g:\Workspace\Cenaero\Git\h2cs-design\env\lib\site-packages\pyomo\opt\base\solvers.py in _solver_error(self, method_name)
    132 + "\n\t".join("%s: %s" % i for i in sorted(self._kwds.items()))
    133 + "\n\t_args: %s" % ( self._args, )
--> 134 + "\n\toptions: %s" % ( self.options, ) )
    135 
    136 

RuntimeError: Attempting to use an unavailable solver.

The SolverFactory was unable to create the solver "bonmin"
and returned an UnknownSolver object.  This error is raised at the point
where the UnknownSolver object was used as if it were valid (by calling
method "solve").

The original solver was created with the following parameters:
	executable: bonmin
	solver_io: lp
	type: bonmin
	_args: ()
	options: {}
```
oemof-solph works well with cbc and ipopt as solver.

Thanks in advance for your help

Best,

Long

**Comment by uvchik** [comment]
HI @longlevan, as far as I know BONMIN and Couenne are solvers for non-linear problems and oemof.solph is only for linear problems.

**Comment by longlevan** [comment]
Thank you very much @uvchik  for your reply. The problem was solved when I used `solver_io ='nl'` instead of `solver_io='lp'` by default in oemof.solph. cbc work with both `solver_io='lp'` and `solver_io='nl'` (the same for ipopt). I'm not sure how relevant to use a NLP or MINLP solver with oemof.solph, just want to test different solvers supported by pyomo. Btw, cbc works much better than others solvers that I tested with oemof.solph.
